src/main/java/appeng/tile/crafting/TileMolecularAssembler.java

```diff
@@ -116,11 +122,7 @@ public boolean pushPattern( final ICraftingPatternDetails patternDetails, final
 	{
 		if( this.myPattern.isEmpty() )
 		{
-			boolean isEmpty = true;
-			for( int x = 0; x < this.inv.getSizeInventory(); x++ )
-			{
-				isEmpty = this.inv.getStackInSlot( x ).isEmpty() && isEmpty;
-			}
+			boolean isEmpty = ItemHandlerUtil.isEmpty(gridInv) && ItemHandlerUtil.isEmpty(patternInv);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') '(' is not followed by whitespace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') ')' is not preceded with whitespace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') '(' is not followed by whitespace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') ')' is not preceded with whitespace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)